doc/man3/OSSL_HPKE_enc.pod

```diff
@@ -0,0 +1,346 @@
+=pod
+
+=head1 NAME
+
+OSSL_HPKE_keygen, 
+OSSL_HPKE_suite_check, OSSL_HPKE_str2suite,
+OSSL_HPKE_expansion,OSSL_HPKE_good4grease,
+OSSL_HPKE_CTX_new, OSSL_HPKE_CTX_free,
+OSSL_HPKE_CTX_set1_psk,
+OSSL_HPKE_CTX_set1_ikme,
+OSSL_HPKE_CTX_set1_senderpriv,
+OSSL_HPKE_CTX_set1_authpriv,
+OSSL_HPKE_CTX_set1_authpub,
+OSSL_HPKE_CTX_get0_seq, OSSL_HPKE_CTX_set1_seq,
+OSSL_HPKE_sender_seal, OSSL_HPKE_recipient_open,
+OSSL_HPKE_CTX_export,
+OSSL_HPKE_enc, OSSL_HPKE_dec
+- Hybrid Public Key Encryption (HPKE) functions
+
+=head1 SYNOPSIS
+
+ #include <crypto/hpke.h>
+
+ OSSL_HPKE_CTX *OSSL_HPKE_CTX_new(int mode, OSSL_HPKE_SUITE suite,
+                                  OSSL_LIB_CTX *libctx, const char *propq);
+
+ void OSSL_HPKE_CTX_free(OSSL_HPKE_CTX *ctx);
+
+ int OSSL_HPKE_CTX_set1_psk(OSSL_HPKE_CTX *ctx,
+                            const char *pskid,
+                            const unsigned char *psk, size_t psklen);
+ int OSSL_HPKE_CTX_set1_senderpriv(OSSL_HPKE_CTX *ctx, EVP_PKEY *privp);
+ int OSSL_HPKE_CTX_set1_authpriv(OSSL_HPKE_CTX *ctx, EVP_PKEY *privp);
+ int OSSL_HPKE_CTX_set1_authpub(OSSL_HPKE_CTX *ctx,
+                                unsigned char *pub,
+                                size_t publen);
+ int OSSL_HPKE_CTX_get0_seq(OSSL_HPKE_CTX *ctx, uint64_t *seq);
+ int OSSL_HPKE_CTX_set1_seq(OSSL_HPKE_CTX *ctx, uint64_t seq);
+ int OSSL_HPKE_CTX_set1_ikme(OSSL_HPKE_CTX *ctx,
+                            const unsigned char *ikme, size_t ikmelen);
+
+ int OSSL_HPKE_sender_seal(OSSL_HPKE_CTX *ctx,
+                           unsigned char *enc, size_t *enclen,
+                           unsigned char *ct, size_t *ctlen,
+                           unsigned char *pub, size_t publen,
+                           const unsigned char *info, size_t infolen,
+                           const unsigned char *aad, size_t aadlen,
+                           const unsigned char *pt, size_t ptlen);
+
+ int OSSL_HPKE_recipient_open(OSSL_HPKE_CTX *ctx,
+                              unsigned char *pt, size_t *ptlen,
+                              const unsigned char *enc, size_t enclen,
+                              EVP_PKEY *recippriv,
+                              const unsigned char *info, size_t infolen,
+                              const unsigned char *aad, size_t aadlen,
+                              const unsigned char *ct, size_t ctlen);
+
+ int OSSL_HPKE_CTX_export(OSSL_HPKE_CTX *ctx,
+                          unsigned char *secret,
+                          size_t secret_len,
+                          const unsigned char *label,
+                          size_t labellen);
+
+ int OSSL_HPKE_keygen(OSSL_LIB_CTX *libctx, const char *propq,
+                      unsigned int mode, OSSL_HPKE_SUITE suite,
+                      const unsigned char *ikm, size_t ikmlen,
+                      unsigned char *pub, size_t *publen, EVP_PKEY **priv);
+
+ int OSSL_HPKE_suite_check(OSSL_HPKE_SUITE suite);
```

Comment:
Not sure the check needs to be public,,,
Maybe the OSSL_HPKE_SUITE should be opaque and have a new method.. You would then just need to do the check once internally when it was created,,,

Reply:
Fairly easy one that in the sense that the action to take after a decision is simple. I'd argue that it's better to distinguish which strings describing suites are valid differently from which suites are supported locally and error out differently based on that. But it's a fairly minor point.